### ISSUE TITLE
CSS is broken in ST 3

### ISSUE THREAD

**Issue by ghost** [sublimehq/Packages#2674]
CSS from master works fine in ST 4, but it is broken in ST 3. When I open any css file in ST 3 with CSS from master from this repo, I get the entire file colored as a css class name. Can someone please confirm the issue and update the style so that it works in ST 3 as well. Thanks.

If nobody can reproduce, I can share my screen recording and I can also bisect it if you want. But save me from effort

**Comment by rwols** [comment]
Please provide a snippet (in text form, optionally a screenshot) that reproduces this behavior.

**Comment by SlavKing98** [comment]
# ST4

![](https://user-images.githubusercontent.com/76569408/103104216-2cd83300-4626-11eb-92f3-1a82bcbc7363.png)

# ST3

![](https://user-images.githubusercontent.com/76569408/103104223-3366aa80-4626-11eb-959a-c0a13fe0d570.png)

```css
.main {
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
}

body {
  margin: 0;
  padding: 0;
}

@media screen and (min-width: 1024px) {
  .container {
    display: -ms-grid;
    display: grid;
    -ms-grid-columns: (1fr)[2];
        grid-template-columns: repeat(2, 1fr);
    -ms-grid-rows: 1fr;
        grid-template-rows: 1fr;
    grid-column-gap: 0px;
    grid-row-gap: 0px;
  }
}

/* Page 1 */
.page-1 {
  background-image: url("/images/skateboarding.jpg");
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
  height: 100vh;
  color: rgba(255, 255, 255, 0.89);
  font-family: "Fjalla One", sans-serif;
  position: relative;
  padding: 1.5rem;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
      -ms-flex-pack: justify;
          justify-content: space-between;
}

@media screen and (min-width: 768px) {
  .page-1 {
    padding: 3rem;
  }
}

@media screen and (min-width: 1024px) {
  .page-1 {
    width: 100%;
    height: 100%;
    -o-object-fit: cover;
       object-fit: cover;
    -o-object-position: 50% 50%;
       object-position: 50% 50%;
  }
}

.page-1__text-container {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-flow: column nowrap;
          flex-flow: column nowrap;
  -webkit-box-pack: justify;
      -ms-flex-pack: justify;
          justify-content: space-between;
  width: 100%;
}

.page-1__heading {
  letter-spacing: 10px;
  text-transform: uppercase;
  text-align: center;
  border-bottom: 10px solid rgba(255, 255, 255, 0.89);
  font-size: 5rem;
  line-height: 8rem;
}

.page-1__sub-heading {
  margin: 0;
  text-transform: uppercase;
  letter-spacing: 2px;
}

.page-1__pg-number {
  letter-spacing: 2px;
}

/* Page 2 */
.page-2 {
  background-color: #f0efeb;
  padding: 1.5rem;
}

@media screen and (min-width: 768px) {
  .page-2 {
    padding: 3rem;
  }
}

@media screen and (min-width: 1024px) {
  .page-2__article {
    padding: 3rem;
  }
}

@media screen and (min-width: 1440px) {
  .page-2__article {
    padding: 5rem;
  }
}

@media screen and (min-width: 1920px) {
  .page-2__article {
    padding: 7rem;
  }
}

@media screen and (min-width: 1024px) {
  .page-2__article-paragraphs-container {
    display: -ms-grid;
    display: grid;
    -ms-grid-columns: (1fr)[2];
        grid-template-columns: repeat(2, 1fr);
    -ms-grid-rows: 1fr;
        grid-template-rows: 1fr;
    grid-column-gap: 30px;
    grid-row-gap: 0px;
  }
}

.page-2__article-paragraph {
  font-family: "Ibarra Real Nova", serif;
  line-height: 1.5;
}

.page-2__sub-heading, .page-2__article-heading, .page-2__article-part, .page-2__pg-number {
  font-family: "Fjalla One", sans-serif;
}

.page-2__article-author {
  font-family: "Ibarra Real Nova", serif;
  color: #eda08d;
}

.page-2__article-heading {
  letter-spacing: 4px;
}

.page-2__article-heading, .page-2__sub-heading, .page-2__article-part {
  text-transform: uppercase;
}

.page-2__sub-heading, .page-2__pg-number {
  letter-spacing: 2px;
}

.page-2__sub-heading {
  margin: 0;
}

span {
  color: #eda08d;
}
/*# sourceMappingURL=style.css.map */
```

**Comment by deathaxe** [comment]
Please note that syntaxes in this repo are always developed against the latest available development built of ST. You won't have any luck in using several syntaxes in ST3 anymore.

Btw.: @SlavKing98  Why did you delete and re-create your account? Are you afraid of being tracked because of your "I cracked ST comment" in https://github.com/sublimehq/Packages/issues/2659#issuecomment-750564340

**Comment by SlavKing98** [comment]
> Please note that syntaxes in this repo are always developed against the latest available development built of ST. You won't have any luck in using several syntaxes in ST3 anymore.

You simply can't do that! ST4 is still in development and is not publicly released yet. Most of the people still use ST3. Also, while ST3 allows you to use unregistered version for free, ST4 doesn't even let you open the editor until you enter a valid license key. Besides that, the only difference is that the git master css definitions use some fancy features that can also be achieved in st3 with a little bit of verboser code. Can you please consider switching back to older syntax definitions version and making it work for ST3 as well. Thanks.

> Why did you delete and re-create your account? Are you afraid of being tracked because of your "I cracked ST comment"

Not really. The jurisdiction system in my country encourages piracy in order to support the domestic software industry. It is perfectly legal here to crack a software. But since github is located in USA, the worst thing that could happen is that github could delete my account if I had had posted something infringing. On the other hand, I haven't posted anything bad; you can't figure out how to crack it just by looking at the screenshot. I posted detailed instructions on my ThePirateBay account though. But the real reason I refreshed my account is because I don't want to keep being chained to the overworld benefits or career since I am probably going to be in a monastery in a couple of years.

**Comment by Ultra-Instinct-05** [comment]
> You simply can't do that! ST4 is still in development and is not publicly released yet. Most of the people still use ST3

I don't see a problem in this. The version of the shipped packages are anyways locked to that particular build so nothing breaks anyways. Cloning this repo and symlinking (or junctioning) to your `Packages` repo is anyways not recommended for normal users (though I do agree that the `README.md` could be updated to indicate that the packages are being developed against ST4). 

> Also, while ST3 allows you to use unregistered version for free, ST4 doesn't even let you open the editor until you enter a valid license key

ST4 alpha (or beta or whatever) builds (at this point) are essentially dev builds, so yes it does require a license. Dev builds requiring licenses has always been the case even for ST3.

 > Besides that, the only difference is that the git master css definitions use some fancy features that can also be achieved in st3 with a little bit of verboser code. Can you please consider switching back to older syntax definitions version and making it work for ST3 as well. Thanks.

That is not possible. ST4 introduces a number of syntax features, new to ST4 only, which won't work or will not be backported to ST3. Switching to older syntax definitions (with more "verboser" code) means loosing on all of the work that has gone to make the definitions more maintainable (not to mention not being able to use all of the new features that is introduced). So I don't think that's happening.

**Comment by deathaxe** [comment]
> ...  while ST3 allows you to use unregistered version for free ...

This is just not correct. You must buy a license for ST3 as well. It's just the developers don't bother you by limiting evaluation time.

> The jurisdiction system in my country encourages piracy in order to support the domestic software industry. 

So why do you bother the global community? Just write your own text editor if you think you can make things soo much better without support from other countries experts! Cracking a software and developing those are two completely different levels of skills, needed.

**Comment by deathaxe** [comment]
> You simply can't do that! ST4 is still in development and is not publicly released yet. 

Yes, we can and this is what this repo is for. It is to prepare syntax definitions shipped with latest dev builts. It is not a common repo to ship 3rd party packages for all sorts of versions of released ST versions.

**Comment by wbond** [comment]
Closing, since this appears to be an issue with an ST3 user trying to use ST4 syntaxes. Should be addressed by #2686
